spikeinterface/qualitymetrics/misc_metrics.py

```diff
@@ -379,17 +378,32 @@ def compute_amplitudes_cutoff(waveform_extractor, peak_sign='neg',
 
     extremum_channels_ids = get_template_extremum_channel(waveform_extractor, peak_sign=peak_sign)
 
+    spike_amplitudes = None
+    if waveform_extractor.is_extension("spike_amplitudes"):
+        amp_calculator = waveform_extractor.load_extension("spike_amplitudes")
+        spike_amplitudes = amp_calculator.get_data(outputs="by_unit")
+    else:
+        warnings.warn(
+            "'spike_amplitude' extension not found. Amplitudes will be computed using waveforms (by default 500), "
+            "which might lead to mis-estimation of amplitude cutoff. Pre-computing amplitudes for all spikes is "
+            "strongly recommended: `spikeinterface.postprocessing.compute_spike_amplitudes(waveform_extractor)`"
+        )
+
     all_fraction_missing = {}
     for unit_id in unit_ids:
-        waveforms = waveform_extractor.get_waveforms(unit_id)
-        chan_id = extremum_channels_ids[unit_id]
-        chan_ind = recording.id_to_index(chan_id)
-        amplitudes = waveforms[:, before, chan_ind]
+        if spike_amplitudes is None:
+            waveforms = waveform_extractor.get_waveforms(unit_id)
+            chan_id = extremum_channels_ids[unit_id]
+            chan_ind = recording.id_to_index(chan_id)
+            amplitudes = waveforms[:, before, chan_ind]
+        else:
+            amplitudes = np.concatenate([spike_amps[unit_id] for spike_amps in spike_amplitudes])
 
         # change amplitudes signs in case peak_sign is pos
         if peak_sign == "pos":
-            amplitudes = -amplitudes
-
+            # for pre-computed amplitudes, values are already positive
+            if np.mean(amplitudes) > 0:
+                amplitudes = -amplitudes
```

Comment:
Here the indentation is not the same.
Maybe we can keep the peak_sign == "pos" stuff in `if spike_amplitudes is None:` case no ?

Reply:
No, you can also compute spike_amplitudes with peak_sign="pos" (https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/postprocessing/spike_amplitudes.py#L136)

Reply:
done!